extensions/google-drive/src/db.ts

```diff
@@ -169,27 +171,79 @@ export const insertFile = (
   const insertStatement = `
       INSERT
         INTO files (name, path, displayPath, fileSizeFormatted, createdAt, updatedAt)
-        VALUES ("${name}", "${path}", "${displayPath}", "${fileSizeFormatted}", "${createdAt.toISOString()}", "${updatedAt.toISOString()}")
+        VALUES ("${name}", "${path}", "${displayPath}", "${fileSizeFormatted}", "${createdAt}", "${updatedAt}")
         ON CONFLICT (path) DO
           UPDATE SET name = EXCLUDED.name, displayPath = EXCLUDED.displayPath, fileSizeFormatted = EXCLUDED.fileSizeFormatted, createdAt = EXCLUDED.createdAt, updatedAt = EXCLUDED.updatedAt;`;
 
   db.run(insertStatement);
 };
 
-export const walkRecursivelyAndSaveFiles = (path: PathLike, db: Database): void => {
-  if (getExcludePaths().includes(path.toLocaleString())) return;
-  saveFilesInDirectory(path, db);
-  getDirectories(path).map((dir) => walkRecursivelyAndSaveFiles(dir, db));
+const listFilesAndInsertIntoDb = async (path: string, db: Database, toast: Toast): Promise<void> => {
+  const updateToastMessage = throttledUpdateToastMessage(toast);
+
+  let totalFiles = 0;
+
+  if (!pathExists(path)) {
+    // TODO: message that path does not exist
+    return;
+  }
```

Comment:
Don't think we need it here. We already check for the `drivePath` to exist and show a message in `index.tsx`.

Reply:
Cool, I'll remove it from here.